pages/History.qml

```diff
@@ -1762,7 +1762,14 @@ Rectangle {
             root.model = appWindow.currentWallet.historyModel;
             root.model.sortRole = TransactionHistoryModel.TransactionBlockHeightRole
             root.model.sort(0, Qt.DescendingOrder);
-            fromDatePicker.currentDate = model.transactionHistory.firstDateTime
+            var count = root.model.rowCount()
+            if (count >0) {
+                //date of the first transaction
+                fromDatePicker.currentDate = root.model.data(root.model.index((count -1), 0), TransactionHistoryModel.TransactionDateRole);
```

Comment:
```suggestion
                fromDatePicker.currentDate = root.model.data(root.model.index((count - 1), 0), TransactionHistoryModel.TransactionDateRole);
```

Either write `count-1` or `count - 1` otherwise it looks confusing :D Same for the `if` above.